sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/resource/AwsResource.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.contrib.trace.aws.resource;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/** Populates the attributes for creating a {@link Resource} that is running in AWS. */
+public abstract class AwsResource {
+
+  /**
+   * Returns a {@link Resource} which is filled with attributes describing the current AWS
+   * environment, e.g., metadata for the instance if the app is running on EC2.
+   */
+  public static Resource create() {
+    return create(new Ec2ResourcePopulator());
+  }
+
+  @VisibleForTesting
+  static Resource create(AwsResource... populators) {
+    Map<String, AttributeValue> resourceAttributes = new LinkedHashMap<>();
```

Comment:
Is it important that we use an ordered-key map implementation here? If so, please add some documentation as to why.

Reply:
I tend to use `LinkedHashMap` when the allocation impact isn't large like these singleton-like objects so order mattering doesn't affect things, if not now then down the line. I suspect `Resource.create` does too for similar reasons. If it's helpful to add that as a comment let me know I'll add it.

Reply:
I do see that Resource.create does wrap the map in a LinkedHashMap implementation, but there's no documentation as to why there, either. If an interface takes a Map, it should definitely assume it won't be ordered, unless there's a very good reason for it.